File: practica2 de auxiliatura/herencia/ejercicio1.py

class Vehiculo:
    def __init__(self, marca, modelo, anio, precio_base):
        self.marca = marca
        self.modelo = modelo
        self.anio = anio
        self.precio_base = precio_base
    def mostrar_info(self):
        print(f"Marca: {self.marca}, Modelo: {self.modelo}, Año: {self.anio}, Precio base: ${self.precio_base}")
    def get_anio(self):
        return self.anio
class Coche(Vehiculo):
    def __init__(self, marca, modelo, anio, precio_base, num_puertas, tipo_combustible):
        super().__init__(marca, modelo, anio, precio_base)
        self.num_puertas = num_puertas
        self.tipo_combustible = tipo_combustible

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Puertas: {self.num_puertas}, Combustible: {self.tipo_combustible}")

    def get_num_puertas(self):
        return self.num_puertas
class Moto(Vehiculo):
    def __init__(self, marca, modelo, anio, precio_base, cilindrada, tipo_motor):
        super().__init__(marca, modelo, anio, precio_base)
        self.cilindrada = cilindrada
        self.tipo_motor = tipo_motor
    def mostrar_info(self):
        super().mostrar_info()
        print(f"Cilindrada: {self.cilindrada}cc, Tipo de motor: {self.tipo_motor}")
vehiculos = [
    Coche("Toyota", "Corolla", 2025, 20000, 4, "Gasolina"),
    Coche("Ford", "Explorer", 2025, 35000, 5, "Híbrido"),
    Moto("Yamaha", "MT-07", 2024, 8000, 689, "2T"),
    Moto("Kawasaki", "Ninja", 2025, 12000, 649, "4T")
]
print("\n--- Información de vehículos ---")
for v in vehiculos:
    v.mostrar_info()
    print()
print("\n--- Coches con más de 4 puertas ---")
for v in vehiculos:
    if isinstance(v, Coche) and v.get_num_puertas() > 4:
        v.mostrar_info()
        print()
print("\n--- Vehículos del año 2025 ---")
for v in vehiculos:
    if v.get_anio() == 2025:
        v.mostrar_info()
        print()

File: practica2deAuxiliaturaPGRA2/src/Herencia/ejercicio1.java
package Herencia;
class Vehiculo {
    protected String marca;
    protected String modelo;
    protected int anio;
    protected double precioBase;

    public Vehiculo(String marca, String modelo, int anio, double precioBase) {
        this.marca = marca;
        this.modelo = modelo;
        this.anio = anio;
        this.precioBase = precioBase;
    }

    public void mostrarInfo() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Año: " + anio + ", Precio base: $" + precioBase);
    }

    public int getAnio() {
        return anio;
    }
}
class Coche extends Vehiculo {
    private int numPuertas;
    private String tipoCombustible;

    public Coche(String marca, String modelo, int anio, double precioBase, int numPuertas, String tipoCombustible) {
        super(marca, modelo, anio, precioBase);
        this.numPuertas = numPuertas;
        this.tipoCombustible = tipoCombustible;
    }

    @Override
    public void mostrarInfo() {
        super.mostrarInfo();
        System.out.println("Puertas: " + numPuertas + ", Combustible: " + tipoCombustible);
    }

    public int getNumPuertas() {
        return numPuertas;
    }
}
class Moto extends Vehiculo {
    private int cilindrada;
    private String tipoMotor;

    public Moto(String marca, String modelo, int anio, double precioBase, int cilindrada, String tipoMotor) {
        super(marca, modelo, anio, precioBase);
        this.cilindrada = cilindrada;
        this.tipoMotor = tipoMotor;
    }

    @Override
    public void mostrarInfo() {
        super.mostrarInfo();
        System.out.println("Cilindrada: " + cilindrada + "cc, Tipo de motor: " + tipoMotor);
    }
}
public class ejercicio1 {
    public static void main(String[] args) {
        Vehiculo[] vehiculos = {
            new Coche("Toyota", "Corolla", 2025, 20000, 4, "Gasolina"),
            new Coche("Ford", "Explorer", 2025, 35000, 5, "Híbrido"),
            new Moto("Yamaha", "MT-07", 2024, 8000, 689, "2T"),
            new Moto("Kawasaki", "Ninja", 2025, 12000, 649, "4T")
        };

        System.out.println("\n--- Información de vehículos ---");
        for (Vehiculo v : vehiculos) {
            v.mostrarInfo();
            System.out.println();
        }

        System.out.println("\n--- Coches con más de 4 puertas ---");
        for (Vehiculo v : vehiculos) {
            if (v instanceof Coche && ((Coche) v).getNumPuertas() > 4) {
                v.mostrarInfo();
                System.out.println();
            }
        }

        System.out.println("\n--- Vehículos del año 2025 ---");
        for (Vehiculo v : vehiculos) {
            if (v.getAnio() == 2025) {
                v.mostrarInfo();
                System.out.println();
            }
        }
    }
}
